org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/InitHandler.java

```diff
@@ -108,6 +110,10 @@ InitializeResult initialize(InitializeParams param) {
 			logInfo("No workspace folders or root uri was defined. Falling back on " + workspaceLocation);
 			rootPaths.add(workspaceLocation);
 		}
+		List<String> javaImportExclusions = getJavaImportExclusions(initializationOptions);
```

Comment:
Rather than cherry-picking the prefs we need during initialization, can't we just get the whole deal instead? We could instantiate a Preference object out of initializationOptions